### DISCUSSION TITLE
Is it possible to turn off widget editing in default screen?

### DISCUSSION THREAD

**Discussion by pilotltd** [EdgeTX/edgetx#1947] [Q&A]
TX16s - accidentally moving scroll wheel on default screen highlights widgets in editing mode. Why? Surely should only be editable in screen editing pages.

**Comment by pfeerick** [comment]
No, not in edit mode. Moving the roller does highlight widgets, so that you can select them to edit options or make them full screen. You can't change or remove the widget though.

How else would you suggest allowing for non-touch interaction with the widgets?

**Reply by pilotltd** [reply]
They need not, and should not become editable from default screen simply by scrolling the roller. Far too easy to do by accident.  Press scroll button to enable edit mode then enable change makes far more sense. Ideally they should only be editable via screens settings and not via default screen - as in OpenTX

**Reply by pfeerick** [reply]
Er, no, you can't press the scroll button (Enter) since that is used to bring up the carousel/main menu. But perhaps a long press to toggle between view and interaction modes? I'm not saying this can't and shouldn't be improved, it's just not as simple as "just put it in screen settings". 

OpenTX does not have fullscreen or widget/app mode concept, so it's UI conventions are not transferable any more. 

Please keep in mind two things 1) whatever UI changes are made need to work for both touch and non-touch UI, and across several radio interface types - i.e. TX16S vs T16 vs X10 and 2), you still need some easy way to select a widget so you can access fullscreen/app mode. There are several widgets starting to make use of it now, where they will show a minimised ui as a widget, and when you select them and go full screen, they will become fullscreen apps and some are interactive already. Maybe have a look at SoarETX, or EventDemo to get a better idea of how this works.

**Reply by pilotltd** [reply]
Selecting/Editing by long press would likely work.  I'm a stick pincher and the ball of my thumb catches the scroll wheel and moves it easily. It would be even worse in winter when wearing gloves. 
Simply moving a loose scroll wheel to allow editing is impracticable for me. Every time I look at screen one or other widget is highlighted with a flashing border and touching screen disables flashing selection. Wrong way round in my opinion, touch screen to enable selection via scroll wheel would make far more sense. Or - give an option to disable scroll wheel in default screen? Full screen interactive versions of widgets are easily set up and readily accessible via page buttons already and personally I find it's much easier to access than screen presses.

**Reply by JimB40** [reply]
This is hardware issue. To exec widget edit you have to move roller, press it, move again (edit is not default) and press it again.
So it's bit like asking to change layout of car's dashboard because light switch is loose and being swiitched off while turning steering wheel. Although we may consider @pfeerick kind of 'lock mode' when all buttons and Touch interface is blocked, for cases like that.

**Reply by pfeerick** [reply]
No, not quite. Yes, to 'select' edit, you have to do a series of steps, which can easily be done inadvertently. But you also don't put the light switch in a position where it "does" get hit by the steering wheel all the time. The scroll wheel can be easily moved, and if you press it expecting the menu carousel to open, and instead the menu pops asking you what to do with the widget, so when you have to hit the back key (or touch the screen if touch enabled). This is why it really should have been a long ENT to bring up the options, rather than short. i.e. same as it is for the touch... the touch doesn't bring up the menu, it's brought up on release of long press ;) A touch lock mode is different, but is also a feature that has been requested before, and I could see encompassing this also.
